File: packages/server-go/internal/handlers/health.go
package handlers

import (
	"net/http"
	"time"

	"github.com/bpmn-explorer/server-go/pkg/database"
	"github.com/gin-gonic/gin"
)

// HealthCheck returns the health status of the server
func HealthCheck(db *database.Database) gin.HandlerFunc {
	return func(c *gin.Context) {
		status := "ok"
		dbStatus := "unavailable"

		if db.IsAvailable() {
			dbStatus = "connected"
		}

		c.JSON(http.StatusOK, gin.H{
			"status":    status,
			"timestamp": time.Now().Format(time.RFC3339),
			"database":  dbStatus,
		})
	}
}

File: server-nodejs/src/handlers/health.ts
import { Context } from 'koa';
import { Database } from '../pkg/database';
import { Logger } from '../pkg/logger';

export async function healthCheck(ctx: Context, db: Database, _logger: Logger) {
  const isDbAvailable = db.isAvailable();

  ctx.status = 200;
  ctx.body = {
    status: 'ok',
    database: isDbAvailable ? 'connected' : 'unavailable',
    timestamp: new Date().toISOString(),
  };
}
